src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -260,10 +260,17 @@ static void notify_on_write(grpc_tcp* tcp) {
   if (grpc_tcp_trace.enabled()) {
     gpr_log(GPR_INFO, "TCP:%p notify_on_write", tcp);
   }
-  cover_self(tcp);
-  GRPC_CLOSURE_INIT(&tcp->write_done_closure,
-                    tcp_drop_uncovered_then_handle_write, tcp,
-                    grpc_schedule_on_exec_ctx);
+  if (grpc_event_engine_run_in_background()) {
```

Comment:
Since this will be constant once the code has started, can this information be memoized rather than having to continuously call a virtual function?

Reply:
In retrospect, I think my comment here was misguided because this isn't an actual virtual function after all. It lives in the vtbl but is just a boolean, and I hadn't noticed that at first.

Reply:
So I think it should be fine to call grpc_event_engine_run_in_background() in the if condition, right? Given that grpc_event_engine_run_in_background() only accesses a boolean member of a global struct, it should not be very costly.